django/website/dashboard/tests/test_mixins.py

```diff
@@ -0,0 +1,183 @@
+from django.contrib.sessions.middleware import SessionMiddleware
+from django.http.response import Http404
+from django.test.client import RequestFactory
+
+from django_dynamic_fixture import G, N
+import mock
+import pytest
+
+from logframe.models import LogFrame
+
+from ..mixins import OverviewMixin, update_current_logframe
+
+
+@pytest.mark.django_db
+def test_get_logframe_returns_new_logframe_if_none_exists():
+    LogFrame.objects.all().delete()
+
+    overview_mixin = OverviewMixin()
+    overview_mixin.request = mock.Mock(user=mock.Mock(last_viewed_logframe=None))
+    overview_mixin.kwargs = {}
+
+    log_frame = overview_mixin.get_logframe()
+
+    assert LogFrame.objects.get() == log_frame
+
+
+@pytest.mark.django_db
+def test_get_logframe_returns_existing_logframe_where_one_exists():
+    LogFrame.objects.all().delete()
+
+    expected_log_frame = G(LogFrame)
+
+    overview_mixin = OverviewMixin()
+    overview_mixin.request = mock.Mock(user=mock.Mock(last_viewed_logframe=None))
+    overview_mixin.kwargs = {}
+    actual_log_frame = overview_mixin.get_logframe()
+
+    assert expected_log_frame == actual_log_frame
+
+
+@pytest.mark.django_db
+def test_get_logframe_returns_first_logframe_by_default():
+    LogFrame.objects.all().delete()
+
+    expected_log_frame = G(LogFrame, n=2)[0]
+
+    overview_mixin = OverviewMixin()
+    overview_mixin.request = mock.Mock(user=mock.Mock(last_viewed_logframe=None))
+    overview_mixin.kwargs = {}
+    actual_log_frame = overview_mixin.get_logframe()
+
+    assert expected_log_frame == actual_log_frame
+
+
+@pytest.mark.django_db
+def test_get_logframe_gets_current_logframe_slug_from_current_user():
+    request = RequestFactory().get('/')
+    request.user = mock.Mock()
+    SessionMiddleware().process_request(request)
+
+    expected_log_frame = G(LogFrame, n=2)[1]
+    request.user.last_viewed_logframe = expected_log_frame
+
+    overview_mixin = OverviewMixin()
+    overview_mixin.request = request
+    overview_mixin.kwargs = {}
+
+    actual_log_frame = overview_mixin.get_logframe()
+
+    assert expected_log_frame == actual_log_frame
+
+
+@pytest.mark.django_db
+def test_get_logframe_gets_current_logframe_slug_from_kwargs():
+    request = RequestFactory().get('/')
+    request.user = mock.Mock()
+    SessionMiddleware().process_request(request)
+
+    expected_log_frame = G(LogFrame, n=2)[1]
+
+    overview_mixin = OverviewMixin()
+    overview_mixin.request = request
+    overview_mixin.kwargs = {'slug': expected_log_frame.slug}
+
+    actual_log_frame = overview_mixin.get_logframe()
+
+    assert expected_log_frame == actual_log_frame
+
+
+@pytest.mark.django_db
+def test_get_logframe_gets_current_logframe_slug_from_kwargs_in_preference_to_session():
```

Comment:
Maybe 'in_preference_to_last_viewed_log_frame' ? We don't talk about sessions previously, it's strange to introduce this here.